### ISSUE TITLE
Bad error on invalid json

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments-migration-test#1210]
(Original issue 1506 created by None on 2019-04-15T08:44:09.915778+00:00)

I had the following invalid json:
```
slicer": {
        "type": ""
}
```
In sphinx it caused the following error:

```
#!text
# Sphinx version: 1.7.4
# Python version: 2.7.16 (CPython)
# Docutils version: 0.14 
# Jinja2 version: 2.10
# Last messages:
#   
#   looking for now-outdated files...
#   none found
#   pickling environment...
#   done
#   checking consistency...
#   done
#   preparing documents...
#   done
#   writing output... [ 50%] 05_geo-services-guide
# Loaded extensions:
#   sphinx.ext.mathjax (1.7.4) from /usr/local/lib/python2.7/dist-packages/sphinx/ext/mathjax.pyc
#   sphinx.ext.coverage (1.7.4) from /usr/local/lib/python2.7/dist-packages/sphinx/ext/coverage.pyc
#   sphinx.ext.todo (1.7.4) from /usr/local/lib/python2.7/dist-packages/sphinx/ext/todo.pyc
#   sphinx.ext.autodoc (1.7.4) from /usr/local/lib/python2.7/dist-packages/sphinx/ext/autodoc/__init__.pyc
#   sphinx.ext.intersphinx (1.7.4) from /usr/local/lib/python2.7/dist-packages/sphinx/ext/intersphinx.pyc
#   sphinx.ext.graphviz (1.7.4) from /usr/local/lib/python2.7/dist-packages/sphinx/ext/graphviz.pyc
#   alabaster (0.7.10) from /usr/local/lib/python2.7/dist-packages/alabaster/__init__.pyc
#   sphinx.ext.ifconfig (1.7.4) from /usr/local/lib/python2.7/dist-packages/sphinx/ext/ifconfig.pyc
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/sphinx/cmdline.py", line 304, in main
    app.build(args.force_all, filenames)
  File "/usr/local/lib/python2.7/dist-packages/sphinx/application.py", line 331, in build
    self.builder.build_update()
  File "/usr/local/lib/python2.7/dist-packages/sphinx/builders/__init__.py", line 338, in build_update
    'out of date' % len(to_build))
  File "/usr/local/lib/python2.7/dist-packages/sphinx/builders/__init__.py", line 399, in build
    self.write(docnames, list(updated_docnames), method)
  File "/usr/local/lib/python2.7/dist-packages/sphinx/builders/__init__.py", line 436, in write
    self._write_serial(sorted(docnames))
  File "/usr/local/lib/python2.7/dist-packages/sphinx/builders/__init__.py", line 445, in _write_serial
    self.write_doc(docname, doctree)
  File "/usr/local/lib/python2.7/dist-packages/sphinx/builders/html.py", line 602, in write_doc
    self.docwriter.write(doctree, destination)
  File "/usr/local/lib/python2.7/dist-packages/docutils/writers/__init__.py", line 80, in write
    self.translate()
  File "/usr/local/lib/python2.7/dist-packages/sphinx/writers/html.py", line 56, in translate
    self.document.walkabout(visitor)
  File "/usr/local/lib/python2.7/dist-packages/docutils/nodes.py", line 174, in walkabout
    if child.walkabout(visitor):
  File "/usr/local/lib/python2.7/dist-packages/docutils/nodes.py", line 174, in walkabout
    if child.walkabout(visitor):
  File "/usr/local/lib/python2.7/dist-packages/docutils/nodes.py", line 174, in walkabout
    if child.walkabout(visitor):
  File "/usr/local/lib/python2.7/dist-packages/docutils/nodes.py", line 174, in walkabout
    if child.walkabout(visitor):
  File "/usr/local/lib/python2.7/dist-packages/docutils/nodes.py", line 174, in walkabout
    if child.walkabout(visitor):
  File "/usr/local/lib/python2.7/dist-packages/docutils/nodes.py", line 166, in walkabout
    visitor.dispatch_visit(self)
  File "/usr/local/lib/python2.7/dist-packages/docutils/nodes.py", line 1882, in dispatch_visit
    return method(node)
  File "/usr/local/lib/python2.7/dist-packages/sphinx/writers/html.py", line 444, in visit_literal_block
    location=(self.builder.current_docname, node.line), **highlight_args
  File "/usr/local/lib/python2.7/dist-packages/sphinx/highlighting.py", line 148, in highlight_block
    hlsource = highlight(source, lexer, formatter)
  File "/usr/local/lib/python2.7/dist-packages/pygments/__init__.py", line 85, in highlight
    return format(lex(code, lexer), formatter, outfile)
  File "/usr/local/lib/python2.7/dist-packages/pygments/__init__.py", line 64, in format
    formatter.format(tokens, realoutfile)
  File "/usr/local/lib/python2.7/dist-packages/pygments/formatter.py", line 95, in format
    return self.format_unencoded(tokensource, outfile)
  File "/usr/local/lib/python2.7/dist-packages/pygments/formatters/html.py", line 850, in format_unencoded
    for t, piece in source:
  File "/usr/local/lib/python2.7/dist-packages/pygments/formatters/html.py", line 690, in _wrap_div
    for tup in inner:
  File "/usr/local/lib/python2.7/dist-packages/pygments/formatters/html.py", line 708, in _wrap_pre
    for tup in inner:
  File "/usr/local/lib/python2.7/dist-packages/pygments/formatters/html.py", line 727, in _format_lines
    for ttype, value in tokensource:
  File "/usr/local/lib/python2.7/dist-packages/pygments/lexer.py", line 191, in streamer
    for _, t, v in self.get_tokens_unprocessed(text):
  File "/usr/local/lib/python2.7/dist-packages/pygments/lexer.py", line 654, in get_tokens_unprocessed
    statetokens = tokendefs[statestack[-1]]
IndexError: list index out of range
```

**Comment by Anteru** [comment]
(Original issue 1506 last updated on 2019-05-06T05:44:43.461754+00:00)

(Issue automaticaly closed due to status in Bitbucket: resolved)

**Comment by Anteru** [comment]
(Original comment by birkenfeld on 2019-05-06T05:42:19.445312+00:00)

Never pop all states from the stack, even if a lexer wants to.

fixes #1506

→ <<cset 5122f0c49fd0>>

**Comment by Anteru** [comment]
(Original comment by birkenfeld on 2019-05-06T05:44:43.408058+00:00)

Never pop all states from the stack, even if a lexer wants to.

fixes #1506

→ <<cset b33cd160cc19>>
